### PR TITLE
Adding version 2019_05_15

### PR DESCRIPTION
**Pull request by shdavido**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿{
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-123456789098",
+    "api-version": "2019-05-15",
+    "location": "wus",
```
Comment (zikalino) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json]: does "wus" work as a location?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-123456789098",
+    "api-version": "2019-01-21",
```
Comment (nschonni) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json]: Looks like the version numbers got reverted to the previous version

**Comment by sergey-shandar** [comment]
@shdavido are you still working on this PR?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json (hunk 1 above)
Comment (shdavido) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json]: changed to "West US"

**Comment by shdavido** [comment]
Hi,

Yes, sorry for the amount of commits it’s the first time I’m changing this project.

I have the remaining errors in the automatic checks:

1) (ModelValidation)Additional properties not allowed:
DataConnections_dataConnectionValidation

2) (ModelValidation)expected type object but found type string:
DataConnections_CheckNameAvailability

3) (lintdif) Properties of a PATCH request body must not be required. PATCH operation: 'DataConnections_Update' Model Definition: 'DataConnection' Property: 'kind'
otential new ARM errors
========================
R2016 - PatchBodyParametersSchema
Properties of a PATCH request body must not be required. PATCH operation: 'DataConnections_Update' Model Definition: 'DataConnection' Property: 'kind'
  at specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/kusto.json:2047


4) (lintdif)Warnings:
  Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. Property: enableDiskEncryption
  Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. Property: isEnabled

As for:

  1.  Changing this will cause us to do a breaking change and a lot of work (it’s exist in earlier versions)
  2.  I don’t sure what exactly it points to, but I think it can only cause a breaking change and a lot of work (it’s exist in earlier versions).
  3.  AFAIK @Ravit Dennis<mailto:radennis@microsoft.com> told me the same as 1,2 about that.
  4.  We already have implanatiation in our backend that is in use, that using those as Booleans, so changing those also can make a lot of work.

I’ll ask for approval although having the listed errors and warnings above, if you have any objections please tell me.

Thanks,
Shahar

**Comment by shdavido** [comment]
Hi,

Yes, sorry for the amount of commits it’s the first time I’m changing this project.

I have the remaining errors in the automatic checks:

1) (ModelValidation)Additional properties not allowed:
DataConnections_dataConnectionValidation

2) (ModelValidation)expected type object but found type string:
DataConnections_CheckNameAvailability

3) (lintdif) Properties of a PATCH request body must not be required. PATCH operation: 'DataConnections_Update' Model Definition: 'DataConnection' Property: 'kind'
otential new ARM errors
========================
R2016 - PatchBodyParametersSchema
Properties of a PATCH request body must not be required. PATCH operation: 'DataConnections_Update' Model Definition: 'DataConnection' Property: 'kind'
  at specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/kusto.json:2047


4) (lintdif)Warnings: 
  Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. Property: enableDiskEncryption
  Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. Property: isEnabled

As for:
1)	Changing this will cause us to do a breaking change and a lot of work (it’s exist in earlier versions)
2)	I don’t sure what exactly it points to, but I think it can only cause a breaking change and a lot of work (it’s exist in earlier versions).
3)	AFAIK @Ravit Dennis told me the same as 1,2 about that.
4)	We already have implanatiation in our backend that is in use, that using those as Booleans, so changing those also can make a lot of work.

I’ll ask for approval although having the listed errors and warnings above, if you have any objections please tell me.

Thanks,
Shahar

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json (hunk 2 above)
Comment (shdavido) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoClustersCheckNameAvailability.json]: I fixed it

**Comment by zikalino** [comment]
@shdavido for lintdiff etc, could you add suppressions in the readme file? otherwise I won't be able to merge.

**Comment by zikalino** [comment]
@sergey-shandar I have reassigned this one to myself. I hope you're ok with that :-)

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoDataConnectionsCheckNameAvailability.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-123456789098",
+    "api-version": "2019-05-15",
+    "clusterName": "kustoProd",
+    "databaseName": "KustoDatabase8",
+    "dataConnectionName": "DataConnections8",
```
Comment (zikalino) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-05-15/examples/KustoDataConnectionsCheckNameAvailability.json]: should be:
dataConnectionName: {
 "name":
 "type":
}

**Review by zikalino — APPROVED** [review]

**Pull request merged**